File: 算法/9. 回文数/IsPalindrome.go
package ___回文数

func isPalindrome(x int) bool {
	if x < 0 || (x > 0 && x%10 == 0) {
		return false
	}
	result := 0
	temp := x
	for {
		if temp == 0 {
			break
		}
		result = result*10 + temp%10
		temp = temp / 10
	}
	return result == x
}

File: 算法/9. 回文数/IsPalindrome.java
public class IsPalindrome {

    class Solution {
        public boolean isPalindrome(int x) {
            if (x < 0 || (x != 0 && x % 10 == 0)) {
                return false;
            }
            int temp = x;
            int num = 0;
            while(temp > 0) {
                num = num * 10 + temp % 10;
                temp = temp / 10;
            }
            return num == x;
        }
    }
}
